strings/ctype-bin.c

```diff
@@ -291,9 +291,15 @@ int my_wc_mb_bin(CHARSET_INFO *cs __attribute__((unused)),
 void my_hash_sort_bin(CHARSET_INFO *cs __attribute__((unused)),
                       const uchar *key, size_t len,ulong *nr1, ulong *nr2)
 {
-  const uchar *end = key + len;
-  ulong tmp1= *nr1;
-  ulong tmp2= *nr2;
+  const uchar *end;
+  ulong tmp1;
+  ulong tmp2;
+
+  if (key == NULL || len == 0)
+    return;
+  end = key + len;
+  tmp1= *nr1;
+  tmp2= *nr2;
```

Comment:
No, should be assertion `key != NULL`, another assertion if `len == 0` (2 assertions is easier to diagnose). No need to forward declare tmp1, tmp2.